### PR TITLE
AEJ-2276 - Test cases for Input validation written for IntelDeflator and PairHMM classes on PR#123

### PR DESCRIPTION
**Pull request by swepnadoss**
Implemented additional test cases to address Input validation changes for IntelDeflator and PairHMM classes on 
https://github.com/Intel-HLS/GKL/pull/123

### CODE REVIEW

**Review by Kmannth — APPROVED** [review]
These look like good Null pointer tests to me.

**Review by Kmannth — APPROVED** [review]
These look like good tests to me.

**Comment by Kmannth** [comment]
@SnehalA can you also review?

File: src/test/java/com/intel/gkl/compression/InflaterUnitTest.java [hunk 1]
```diff
@@ -20,12 +20,99 @@
...
+    @Test (expectedExceptions = {NullPointerException.class})
+    public void testInvalidInputsBufferForInflate1() {
+        final byte[] inputbuffer = new byte[(int)inputBytes];
+
+        try {
+            inflaterInputTest.reset();
+            inflaterInputTest.setInput(inputbuffer, 0, inputbuffer.length);
+            inflaterInputTest.inflate(null, 0, inputbuffer.length);
+            inflaterInputTest.end();
+        } catch (java.util.zip.DataFormatException e) {
```
Comment (SnehalA) [inline src/test/java/com/intel/gkl/compression/InflaterUnitTest.java:65]: Why we have try-catch here but not in other places?

File: src/test/java/com/intel/gkl/pairhmm/PairHmmUnitTest.java [hunk 2]
```diff
@@ -22,6 +22,60 @@
...
+
+    @Test (expectedExceptions = {NullPointerException.class})
+    public void testInvalidInputsLikelihoodArrayForComputeLikelihoods() {
+        final IntelPairHmm pairHmm = new IntelPairHmm();
+
+        final PairHMMNativeArguments args = new PairHMMNativeArguments();
+        args.maxNumberOfThreads = 1;
+        args.useDoublePrecision = false;
+
+        pairHmm.initialize(args);
```
Comment (SnehalA) [inline src/test/java/com/intel/gkl/pairhmm/PairHmmUnitTest.java:69]: Do we plan on checking constructors inputs?

File: src/test/java/com/intel/gkl/compression/InflaterUnitTest.java [hunk 3]
```diff
@@ -20,12 +20,99 @@
...
+
+    @Test (expectedExceptions = {IllegalArgumentException.class})
+    public void testInvalidInputsOffsetForInflate1() {
+        final byte[] inputbuffer = new byte[(int)inputBytes];
+        final byte[] outputbuffer = new byte[(int)inputBytes];
+
+        try {
+            inflaterInputTest.reset();
+            inflaterInputTest.setInput(inputbuffer, 0, inputbuffer.length);
+            inflaterInputTest.inflate(outputbuffer, -1, inputbuffer.length);
```
Comment (SnehalA) [inline src/test/java/com/intel/gkl/compression/InflaterUnitTest.java:78]: Should we be checking that inputbuffer and outputBuffer to be same?

**Review by swepnadoss — COMMENTED** [review]
@SnehalA I've addressed all your review comments.

**Review by SnehalA — APPROVED** [review]

**Pull request merged**